slack-message.c

```diff
@@ -9,6 +10,72 @@
 #include "slack-conversation.h"
 #include "slack-message.h"
 
+GString *slack_get_thread_color(const char *ts) {
+	// Produce a color that works against white background by the following
+	// algorithm:
+	//
+	// 1. Pick a pseudo-random number by seeding it with ts, IOW it is
+	// deterministic.
+	//
+	// 2. Pick a 24-bit RGB value, but throw away the the original highest
+	// bit of each byte, resulting in a range from 000000 to 7f7f7f.
+	//
+	// 3. Pick exactly one base color that will receive its highest bit, or
+	// possibly none of them.
+	//
+	// This gives quite a random color, but often with a dominant RGB
+	// component, never white, and deterministic from ts.
+
+	unsigned int seed = slack_parse_time_str(ts);
+	char *dot = strchr(ts, '.');
+	if (dot)
+		seed ^= atol(dot + 1);
+
+	GString *tmp = g_string_sized_new(7);
+
+	unsigned int r = rand_r(&seed);
```

Comment:
This is a slightly strange use of rand. Maybe `g_direct_hash` or even `g_str_hash` on the original ts would be a better fit.

Reply:
Good idea, also fixed!